private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/maven/MavenResolver.java

```diff
@@ -0,0 +1,423 @@
+                  Coordinates remapped = remappings.getOrDefault(from, from);
+                  toReturn.addNode(remapped);
+
+                  actualNode.getChildren().stream()
+                      .map(DependencyNode::getArtifact)
+                      .map(this::amendArtifact)
+                      .map(MavenCoordinates::asCoordinates)
+                      .map(c -> remappings.getOrDefault(c, c))
+                      .forEach(
+                          to -> {
+                            toReturn.addNode(to);
+                            toReturn.putEdge(remapped, to);
+                          });
+                }));
+    directDependencies.forEach(node -> node.accept(collector));
+
+    return ImmutableGraph.copyOf(toReturn);
+  }
+
+  private static DependencyNode getDependencyNode(DependencyNode node) {
+    Map<?, ?> data = node.getData();
+    if (data != null) {
+      Object winner = data.get(ConflictResolver.NODE_DATA_WINNER);
+      if (winner instanceof DependencyNode) {
+        return (DependencyNode) winner;
+      }
+    }
+    return node;
+  }
+
+  private Artifact amendArtifact(Artifact artifact) {
+    // If someone has depended on an aggregating pom, the `type` or `extension` will be `pom`.
+    // However, in `rules_jvm_external`, we pretend that these are actually `jar` files, and
+    // then make things up from there.
+    if (!"pom".equals(artifact.getExtension())) {
+      return artifact;
+    }
+
+    return new DefaultArtifact(
+        artifact.getGroupId(),
+        artifact.getArtifactId(),
+        artifact.getClassifier(),
+        null,
+        artifact.getVersion());
+  }
+
+  private DefaultRepositorySystemSession prepareSession(
+      RepositorySystem system,
+      DependencyManager dependencyManager,
+      RepositoryListener listener,
+      Path localCache) {
+    DefaultRepositorySystemSession session = MavenRepositorySystemUtils.newSession();
+
+    LocalRepository localRepository = new LocalRepository(localCache.toAbsolutePath().toString());
+    session.setLocalRepositoryManager(system.newLocalRepositoryManager(session, localRepository));
+
+    session.setDependencyManager(dependencyManager);
+
+    session.setDependencyTraverser(new StaticDependencyTraverser(true));
+
+    session.setRepositoryListener(listener);
+
+    // Only resolve from repos that we have been asked to use.
+    session.setIgnoreArtifactDescriptorRepositories(true);
+
+    Map<Object, Object> configProperties = new HashMap<>();
+    configProperties.putAll(System.getProperties());
+
+    // Use the breadth-first dependency collector. This allows us to work out the dependency graph
+    // in parallel.
+    // This only applies to the step where we calculate the dependency graph, not the actual
+    // downloading of those
+    // dependencies (which is handled by the `Downloader`.
+    configProperties.put("aether.dependencyCollector.impl", "bf");
```

Comment:
Mind adding to the comment that these configuration options come from [https://github.com/apache/maven-resolver/blob/master/src/site/markdown/configuration.md](https://github.com/apache/maven-resolver/blob/master/src/site/markdown/configuration.md)